lib/segment/src/index/field_index/full_text_index/inverted_index.rs

```diff
@@ -56,28 +57,40 @@ impl ParsedQuery {
     }
 }
 
-#[derive(Default)]
-pub struct InvertedIndex {
-    postings: Vec<Option<PostingList>>,
-    pub vocab: HashMap<String, TokenId>,
-    pub point_to_docs: Vec<Option<Document>>,
-    pub points_count: usize,
+pub enum InvertedIndex {
+    Mutable(MutableInvertedIndex),
+    Immutable(ImmutableInvertedIndex),
 }
 
 impl InvertedIndex {
-    pub fn new() -> InvertedIndex {
-        Default::default()
+    pub fn new(is_appendable: bool) -> InvertedIndex {
+        if is_appendable {
+            InvertedIndex::Mutable(MutableInvertedIndex::default())
+        } else {
+            InvertedIndex::Immutable(ImmutableInvertedIndex::default())
+        }
     }
 
     pub fn document_from_tokens(&mut self, tokens: &BTreeSet<String>) -> Document {
+        let vocab = match self {
+            InvertedIndex::Mutable(index) => &mut index.vocab,
+            InvertedIndex::Immutable(index) => &mut index.vocab,
+        };
+        Self::document_from_tokens_impl(vocab, tokens)
+    }
+
+    fn document_from_tokens_impl(
```

Comment:
Don't like this private method but I use it in two places - in `document_from_tokens` and in `build_index`